resource_planning/wizard/allocate_resource_wizard.py

```diff
@@ -82,13 +82,22 @@ def search_resources(self):
         }
 
     @api.multi
-    def book_resource(self):
-        if self.resources:
-            self.resources.allocate_resource(
-                self.allocation_type,
-                self.date_start,
-                self.date_end,
-                self.partner_id,
-                self.location,
-                self.date_lock,
+    def book_resources(self):
+        self.ensure_one()
+        for resource in self.resources:
+            # cf resource.check_availabilities(
+            #  this function returns resource.available records
```

Comment:
`resource.check_availabilities()` returns `resource.resource` records. i’m not sure i understand the comment.

Reply:
> # cf resource.check_availabilities(
> #  this function returns resource.available records
> #  nothing was done with it here

I think that comment belongs to the commit message. I'm explaining why I create the allocation directly rather than using check_availabilities.

> hope book_resources is called right after search_resources

This function is called from a button shown after the button search_resources was clicked. We therefore hope the allocated resource is still free. It's a bit risky but I stopped refactoring.